### PR TITLE
Sync: Critical Path Tests

### PR DESCRIPTION
**Pull request by malmstein**
Task/Issue URL: https://app.asana.com/0/1174433894299346/1205433252919607

### Description
This PR adds the e2e test to verify that data can be synced across accounts.

### Steps to test this PR
Run maestro test locally, replacing `${CODE}` in `action_add_new_device` with a valid recovery code from the test account

https://github.com/duckduckgo/Android/actions/runs/7646940140/job/20836834057

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206229586525308

### CODE REVIEW

**Comment by malmstein** [comment]
Current dependencies on/for this PR:
* `develop`
  * **PR #4031** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4031?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/4031?utm_source=stack-comment).

**Comment by malmstein** [comment]
@cmonfortep  this is finally passing in https://github.com/duckduckgo/Android/actions/runs/7646940140/job/20836834057

**Review by cmonfortep — COMMENTED** [review]
Just discussed offline:
- move action yml files to an specific folder
- comment expectations around where user should be for the action to success (e.g: browsertabfragment)
- format files

File: .maestro/sync_flows/sync_data.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+
```
Comment (cmonfortep) [inline .maestro/sync_flows/sync_data.yaml:1]: I see other tests have `# Test Definition:`. Can we add a link here as well? I think is a great idea.

**Review by cmonfortep — APPROVED** [review]
LGTM Thanks!

**Pull request merged**